pkgs/by-name/sh/shapr3d/package.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  fetchurl,
+  stdenv,
+  undmg,
+}:
+let
+  inherit (stdenv.hostPlatform) system;
+  unsupported = throw "Shapr3D doesn't support ${system}";
+  identifier = "5.910.0.9193";
+  code =
+    {
+      aarch64-darwin = "aCxANPRSe/biEZVL2fHOHt1h7b2vMDwafuaU9NHoy5Q=";
+    }
+    .${system} or unsupported;
+  url =
+    {
+      aarch64-darwin = fetchurl {
+        url = "https://download.shapr3d.com/mac/Shapr3D-${identifier}.dmg";
+        sha256 = code;
```

```suggestion
        hash = "";
```
`hash` attr is now preferred over `sha*`, and it also needs to be generated by Nix itself. Also no need to have a separate abstraction (`code`) for hashes, just have both URL and hash together